### PR TITLE
Refactor structure

### PR DESCRIPTION
**Pull request by terryluan12**

### CODE REVIEW

**Review by nicholasshort — APPROVED** [review]
Looks good!

File: genrify/datasources/datasources.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import opendatasets as od
+import shutil
+
+def download_datasets():
+    # download gtzan
+    od.download_kaggle_dataset("https://www.kaggle.com/datasets/andradaolteanu/gtzan-dataset-music-genre-classification", "./")
+    shutil.rename('gtzan-dataset-music-genre-classification', 'Data')
```
Comment (jack-jintao-huang) [inline genrify/datasources/datasources.py]: The file structure here is 
gtzan-dataset-music-genre-classification - 
    |---- Data
    |----|---- genre1
    etc.

So we can't just rename the data. Please have a look at what I currently done

File: genrify/preprocessing/preprocessing.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    Args:
+        data_dir (str): path to data directory. Assume that the data_dir has the following structure:
+            data_dir
+            ├── genre1
+            │   ├── file1.wav
+            │   ├── file2.wav
+            │   └── ...
+    """
+
+    # TODO: Add function call to split audio files into 3 second clips
```
Comment (jack-jintao-huang) [inline genrify/preprocessing/preprocessing.py]: Please run the `split_into_3_seconds()` here and also set the `data_dir` for the rest of the processing functions

**Comment by jack-jintao-huang** [comment]
Should we rename the `genrify` folder to src?

**Comment by terryluan12** [comment]
> Should we rename the `genrify` folder to src?

Yep, we can! It's not that big of a difference, [just different package layouts](https://py-pkgs.org/04-package-structure.html)

**Comment by jack-jintao-huang** [comment]
> > Should we rename the `genrify` folder to src?
> 
> Yep, we can! It's not that big of a difference, [just different package layouts](https://py-pkgs.org/04-package-structure.html)

Oh interesting. Today I learnt that python packages. Thanks for sharing

File: genrify/datasources/datasources.py (hunk 1 above)
Comment (terryluan12) [reply genrify/datasources/datasources.py]: The issue should be resolved now. We are also now downloading from osf which doesn't require a kaggle account

**Review by jack-jintao-huang — APPROVED** [review]
lgtm

File: genrify/datasources/datasources.py (hunk 1 above)
Comment (jack-jintao-huang) [reply genrify/datasources/datasources.py]: oh I see. Awesome

**Pull request merged**